SimpleWAWS/Code/BackgroundQueueManager.cs

```diff
@@ -158,43 +168,106 @@ private void HandleBackgroundOperation(BackgroundOperation operation)
+                });
+            }
+            catch (Exception e)
+            {
+                SimpleTrace.Diagnostics.Fatal(e, "MaintainResourceGroupLists error, Count {Count}", Interlocked.Increment(ref _maintainResourceGroupListErrorCount));
+            }
 
         }
 
-        private void OnTimerElapsed(object state)
+        private void OnLogQueueStatsTimerElapsed(object state)
         {
             try
             {
                 _jobHost.DoWork(() =>
                 {
-                    MaintainResourceGroupLists();
-                    _logCounter++;
-                    if (_logCounter % 5 == 0)
+                    LogQueueStatistics();
+                });
+            }
+            catch (Exception e)
+            {
+                SimpleTrace.Diagnostics.Fatal(e, "LogQueueStatistics error, Count {Count}", Interlocked.Increment(ref _maintainResourceGroupListErrorCount));
+            }
+        }
+
+        private void OnCleanupSubscriptionsTimerElapsed(object state)
+        {
+            try
+            {
+                _jobHost.DoWork(async () =>
+                {
+                    if (!this.BackgroundInternalOperations.Any((a) => a.Value.Type == OperationType.SubscriptionLoad))
                     {
-                        //log only every 5 minutes
-                        LogQueueStatistics();
-                        _logCounter = 0;
+                        foreach (var subcription in await CsmManager.GetSubscriptions())
+                        {
+                            {
+                                SubscriptionCleanup(new Subscription(subcription));
+                            }
+                        }
                     }
                 });
             }
             catch (Exception e)
             {
-                SimpleTrace.Diagnostics.Fatal(e, "MainTainResourceGroupLists error, Count {Count}", Interlocked.Increment(ref _maintainResourceGroupListErrorCount));
+                SimpleTrace.Diagnostics.Fatal(e, "CleanupSubscriptions error, Count {Count}", Interlocked.Increment(ref _maintainResourceGroupListErrorCount));
             }
-        }
 
-        private void MaintainResourceGroupLists()
+        }
+        private async Task MaintainResourceGroupLists()
         {
             var resources = this.ResourceGroupsInUse
                 .Select(e => e.Value)
                 .Where(rg => (int)rg.TimeLeft.TotalSeconds == 0);
 
             foreach (var resource in resources)
+            {
                 DeleteResourceGroup(resource);
-            //TODO:Add a way replenish leaking resourcegroups
+            }
+
+            //Delete any duplicate resourcegroups in same subscription loaded in the same region
+            //or create any missing resourcegroups in a region
+            IList<Tuple<string,MakeSubscriptionFreeTrialResult>> subscriptionStates = new List<Tuple<string, MakeSubscriptionFreeTrialResult>>();
```

Comment:
Good point I will have the remove from free queue logic put in. Users will get an exception and asked to try again and it should work the next time since the RG is now out of the free queue. But this is a bad experience  .